### PR TITLE
Add rmw_get_gid_for_client & tests

### PR DESCRIPTION
**Pull request by ihasdapie**
See: https://github.com/ros2/rmw/pull/327

Signed-off-by: Brian Chen <brian.chen@openrobotics.org>

### CODE REVIEW

File: test_rmw_implementation/test/test_unique_identifiers.cpp [hunk 1]
```diff
@@ -113,6 +119,41 @@ TEST_F(CLASSNAME(TestUniqueIdentifierAPI, RMW_IMPLEMENTATION), get_gid_with_bad_
...
+  const char * implementation_identifier = cli->implementation_identifier;
+  cli->implementation_identifier = "not-an-rmw-implementation-identifier";
+  ret = rmw_get_gid_for_client(cli, &gid);
+  cli->implementation_identifier = implementation_identifier;
+  EXPECT_EQ(RMW_RET_INCORRECT_RMW_IMPLEMENTATION, ret);
+  rmw_reset_error();
+  gids_are_equal = false;
+  ret = rmw_compare_gids_equal(&expected_gid, &gid, &gids_are_equal);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  EXPECT_TRUE(gids_are_equal);
```
Comment (ivanpauno) [inline test_rmw_implementation/test/test_unique_identifiers.cpp:154]: why?
`gid` should never have changed if the function called change, so it should still be equal to the original `expected_gid`

File: test_rmw_implementation/test/test_unique_identifiers.cpp [hunk 2]
```diff
@@ -221,3 +262,52 @@ TEST_F(CLASSNAME(TestUniqueIdentifiersForMultiplePublishers, RMW_IMPLEMENTATION)
...
+  ASSERT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+
+  ret = rmw_get_gid_for_client(cli2, &cli2_gid);
+  ASSERT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+
+  bool are_equal = true;
+  ret = rmw_compare_gids_equal(&cli1_gid, &cli2_gid, &are_equal);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  EXPECT_FALSE(are_equal);
+}
```
Comment (ivanpauno) [inline test_rmw_implementation/test/test_unique_identifiers.cpp:313]: I'm not sure this is something we should we testing here, but it's okay to add it

File: test_rmw_implementation/test/test_unique_identifiers.cpp (hunk 1 above)
Comment (jacobperron) [reply test_rmw_implementation/test/test_unique_identifiers.cpp:154]: That's correct. It is still equal to the original, hence we expect a true value. This test is checking that `gid` is not modified whenever we make a bad call to the getter. I think this logic is basically copied over from the tests for the publisher getter.

File: test_rmw_implementation/test/test_unique_identifiers.cpp (hunk 2 above)
Comment (jacobperron) [reply test_rmw_implementation/test/test_unique_identifiers.cpp:313]: Yeah, it seems a bit odd to me too. I believe this was added to copy the test for the publisher gid in this file.

**Comment by jacobperron** [comment]
I forgot to push my local changes, which address some bugs. @ivanpauno please take another look.

File: test_rmw_implementation/test/test_unique_identifiers.cpp (hunk 1 above)
Comment (ivanpauno) [reply test_rmw_implementation/test/test_unique_identifiers.cpp:158]: ah sorry, I misread the code

**Review by ivanpauno — APPROVED** [review]
LGTM!

File: test_rmw_implementation/test/test_unique_identifiers.cpp (hunk 2 above)
Comment (ihasdapie) [reply test_rmw_implementation/test/test_unique_identifiers.cpp]: Yeah, these methods (and the comment from the previous review) were just copied over to be symmetrical with the publisher gid tests

**Pull request merged**